packages/sitecore-jss-react-native/src/components/sharedTypes.ts

```diff
@@ -1,3 +1,3 @@
-import { Component } from 'react';
+import { ComponentType } from 'react';
 
-export type ComponentFactory = (componentName: string) => Component;
+export type ComponentFactory = (componentName: string) => ComponentType | null;
```

Comment:
Seems like a breaking change, but it's valid, because Component can't be Class or FunctionComponent